src/ext/java/org/opentripplanner/ext/siri/SiriTimetableSnapshotSource.java

```diff
@@ -1136,15 +1092,14 @@ private Result<?, UpdateError> addTripToGraphAndBuffer(
     final TripPattern pattern = tripPatternCache.getOrCreateTripPattern(
       stopPattern,
       trip,
-      transitModel,
       serviceDate
     );
 
     // Add new trip times to the buffer and return success
     var result = buffer.update(pattern, updatedTripTimes, serviceDate);
 
     // Add TripOnServiceDate to buffer if a dated service journey id is supplied in the SIRI message
-    addTripOnServiceDateToBuffer(trip, serviceDate, estimatedVehicleJourney, feedId);
+    addTripOnServiceDateToBuffer(trip, serviceDate, estimatedVehicleJourney, entityResolver);
```

Comment:
What if we are using `FramedVehicleJourneyRef`? There will be no DSJ id. How can we add a trip then?

Reply:
Exactly, either you need to have the DSJ in the plan data, or in `EstimatedVehicleJourney -> DatedVehicleJourneyRef`. I added support to fetch the DSJ id from the scheduled trip via the `FramedVehicleJourneyRef`.

Reply:
So, it will resolve DSJ based on SJ + operating day date?

Reply:
Exactly